### DISCUSSION TITLE
How to make this keybinding to open a split with another shell type?

### DISCUSSION THREAD

**Discussion by eggbean** [microsoft/terminal#14056] [Q&A] [answered]
I have four different shells 1, 2, 3, and 4.

Say I want to open shell n in a split with the current one, just like you can by clicking it in the drop-down list while pressing `alt`.

I want to add a key binding for that.

       {
            "command": 
            {
                "action": "splitPane",
                "split": "???",
                "splitMode": "???"
            },
            "keys": "ctrl+alt+shift+n"
        },

I expect that I will need to replace the `n` above with numbers and make four different entries, but how do I do the rest?

**Comment by DHowett** [comment, marked as answer]
You can accomplish this in a couple of ways! Here's how I'd approach it.

## Using indexed profiles

This way, you're at the mercy of your profile ordering.

Each `splitPane` action contains an `index`:

```json
{
    "command": {
        "action": "splitPane",
        "index": 0
    },
    "keys": "ctrl+alt+shift+1"
},
// ...
```

If you hold down <kbd>Alt</kbd> while you open the Settings, you can get at the built-in actions for `newTab` and copy those!

## Using named profiles

This one doesn't put you at the whims of the profile order. Each action contains a `profile`, which is either a GUID or a name:

```json
{
    "command": {
        "action": "splitPane",
        "profile": "Windows PowerShell"
    },
    "keys": "ctrl+alt+shift+1"
},
{
    "command": {
        "action": "splitPane",
        "profile":  "{0caa0dad-35be-5f56-a8ff-afceeeaa6101}" // Command Prompt
    },
    "keys": "ctrl+alt+shift+2"
},
// ...
```

Using the name in this one is _slightly_ worse because if you rename the profile we won't update it for you. It's a pain (sorry).

**Comment by eggbean** [comment]
@DHowett Thanks a lot. That was easier than I thought. I put in 10 entries for indexes 0-9 like this, so that I don't have anything further to do when I move around or add more shells. I just need to remember the order, like I already do when opening then with `Ctrl+Shift+n`.

        {
            "command": {
                "action": "splitPane",
                "index": 0
            },
            "keys": "ctrl+alt+shift+1"
        },
        {
            "command": {
                "action": "splitPane",
                "index": 1
            },
            "keys": "ctrl+alt+shift+2"
        },

**Reply by DHowett** [reply]
Glad it worked! Thanks!
